### PR TITLE
Crystallizer quality fix

### LINKED ISSUE #65767: Inconsiencies with crystallizer mol discount
## Reproduction:
Making higher quality items in the crystallier is supposed to give a discount up to 85% on the amount of mols required by a gived recipe. Due to how quality loss is calculated only recipies with a total mol requirement above about 2000 mols can reach maximum discount, which makes this impossible to reach on most recipes.

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
fixes #65767 
Makes the crystallizer quality control based on the progression bar control, that way the calculations will be in par of each other

## Why It's Good For The Game
consistency fix

## Changelog

:cl:
fix: crystallizer max quality can now be reached regardless of recipe total moles amount
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/components/gas_recipe_machines/crystallizer.dm [hunk 1]
```diff
@@ -137,11 +137,11 @@
 /obj/machinery/atmospherics/components/binary/crystallizer/proc/heat_calculations()
 	if(	(internal.temperature >= (selected_recipe.min_temp * MIN_DEVIATION_RATE) && internal.temperature <= selected_recipe.min_temp) || \
 		(internal.temperature >= selected_recipe.max_temp && internal.temperature <= (selected_recipe.max_temp * MAX_DEVIATION_RATE)))
-		quality_loss = min(quality_loss + 1.5, 100)
+		quality_loss = min(quality_loss + (MIN_PROGRESS_AMOUNT * 4.5 / (round(log(10, total_recipe_moles * 0.1), 0.01))), 100)
 
 	var/median_temperature = (selected_recipe.max_temp + selected_recipe.min_temp) / 2
 	if(internal.temperature >= (median_temperature * MIN_DEVIATION_RATE) && internal.temperature <= (median_temperature * MAX_DEVIATION_RATE))
-		quality_loss = max(quality_loss - 5.5, -85)
+		quality_loss = max(quality_loss - (MIN_PROGRESS_AMOUNT * 4.5 / (round(log(10, total_recipe_moles * 0.1), 0.01))), -85)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/gas_recipe_machines/crystallizer.dm:144]: If this is just gonna be the same number twice could you just calculate it once? it'll make this code easier to read too

File: code/modules/atmospherics/machinery/components/gas_recipe_machines/crystallizer.dm [hunk 2]
```diff
@@ -137,11 +137,11 @@
 /obj/machinery/atmospherics/components/binary/crystallizer/proc/heat_calculations()
 	if(	(internal.temperature >= (selected_recipe.min_temp * MIN_DEVIATION_RATE) && internal.temperature <= selected_recipe.min_temp) || \
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/gas_recipe_machines/crystallizer.dm:138]:
```suggestion
	if((internal.temperature >= (selected_recipe.min_temp * MIN_DEVIATION_RATE) && internal.temperature <= selected_recipe.min_temp) || \
```
This feels unintentional
